### PR TITLE
Fix: GroupsMixin expiry support

### LINKED ISSUE #43: Fix `GroupsMixin` methods to match API documentation
The expected request path and body for including expiration as described by the API documentation are slightly different from the [notes that we were referencing](https://docs.google.com/document/d/1vActZALSbRwO0HxDvm-YQK37p22SPbL4XuMdcRx-BI8/edit#heading=h.im71uyz2mmt).

## Acceptance criteria
- The implementation matches API documentation for:
   - [ ] [Linking](https://docs.littlepay.io/reference/backoffice/tag/Products/#tag/Products/operation/LinkFundingSourceToConcessionGroup)
   - [ ] [Updating](https://docs.littlepay.io/reference/backoffice/tag/Products/#tag/Products/operation/UpdateConcession)

### PR DESCRIPTION
**Pull request by angela-tran**
Closes #43 

This PR updates the request path and request body for the "update" endpoint and the request body for the "linking" endpoint.

See [comment](https://github.com/cal-itp/littlepay/pull/45#issuecomment-2035569396) below on something else I noticed while making these naming updates.

### CODE REVIEW

File: tests/api/test_groups.py [hunk 1]
```diff
@@ -272,13 +280,13 @@ def test_GroupsMixin_link_concession_group_funding_source_expiry(
     mock_ClientProtocol_post_link_concession_group_funding_source, mocker
 ):
     client = GroupsMixin()
-    mocker.patch.object(client, "_format_concession_expiry", return_value="formatted concession expiry")
+    mocker.patch.object(client, "_format_expiry", return_value="formatted concession expiry")
```
Comment (thekaveman) [inline tests/api/test_groups.py:283]: Just a nit but we might as well get rid of the word `concession` here too

File: tests/api/test_groups.py [hunk 2]
```diff
@@ -272,13 +280,13 @@ def test_GroupsMixin_link_concession_group_funding_source_expiry(
...
 
     result = client.link_concession_group_funding_source("group-1234", "funding-source-1234", datetime.now())
 
     endpoint = client.concession_group_funding_source_endpoint("group-1234")
     mock_ClientProtocol_post_link_concession_group_funding_source.assert_called_once_with(
-        endpoint, {"id": "funding-source-1234", "concession_expiry": "formatted concession expiry"}, dict
+        endpoint, {"id": "funding-source-1234", "expiry": "formatted concession expiry"}, dict
```
Comment (thekaveman) [inline tests/api/test_groups.py:289]: Same nit about `concession`

File: tests/api/test_groups.py [hunk 3]
```diff
@@ -287,13 +295,13 @@ def test_GroupsMixin_update_concession_group_funding_source_expiry(
     mock_ClientProtocol_put_update_concession_group_funding_source, ListResponse_GroupFundingSources, mocker
 ):
     client = GroupsMixin()
-    mocker.patch.object(client, "_format_concession_expiry", return_value="formatted concession expiry")
+    mocker.patch.object(client, "_format_expiry", return_value="formatted concession expiry")
```
Comment (thekaveman) [inline tests/api/test_groups.py:298]: Same nit about `concession`

File: tests/api/test_groups.py [hunk 4]
```diff
@@ -287,13 +295,13 @@ def test_GroupsMixin_update_concession_group_funding_source_expiry(
...
 
     result = client.update_concession_group_funding_source_expiry("group-1234", "funding-source-1234", datetime.now())
 
-    endpoint = client.concession_group_funding_source_endpoint("group-1234")
+    endpoint = client.concession_group_funding_source_endpoint("group-1234", "funding-source-1234")
     mock_ClientProtocol_put_update_concession_group_funding_source.assert_called_once_with(
-        endpoint, {"id": "funding-source-1234", "concession_expiry": "formatted concession expiry"}, ListResponse
+        endpoint, {"expiry": "formatted concession expiry"}, ListResponse
```
Comment (thekaveman) [inline tests/api/test_groups.py:304]: Same nit about `concession`

**Review by thekaveman — CHANGES_REQUESTED** [review]
Really minor nitpicks

**Comment by angela-tran** [comment]
I noticed that documentation says this for the `expiry` field in the request body:

> Can be left empty to indicate that the concession does not expire.
> 
> Note that the time is truncated to the minute.

I think the first sentence there answers one of the questions we had about removing expiration. As for the second sentence, I'm not sure if this is saying that we need to truncate the time when we send it in the request body.

File: tests/api/test_groups.py (hunk 3 above)
Comment (angela-tran) [reply tests/api/test_groups.py]: Removed in https://github.com/cal-itp/littlepay/pull/45/commits/484cf53ab421a1254e28da48bb39364e141b8f8c

File: tests/api/test_groups.py (hunk 2 above)
Comment (angela-tran) [reply tests/api/test_groups.py]: Removed in https://github.com/cal-itp/littlepay/pull/45/commits/484cf53ab421a1254e28da48bb39364e141b8f8c

File: tests/api/test_groups.py (hunk 1 above)
Comment (angela-tran) [reply tests/api/test_groups.py]: Removed in https://github.com/cal-itp/littlepay/pull/45/commits/484cf53ab421a1254e28da48bb39364e141b8f8c

**Comment by thekaveman** [comment]
> I think the first sentence there answers one of the questions we had about removing expiration.

Hmmm leaving it empty to indicate it doesn't expire seems different from leaving it empty to remove an existing expiry. We have that question out to Littlepay so hopefully they'll get back to us.

> Note that the time is truncated to the minute.

I interpret this to mean no matter how granular a datetime we send, Littlepay will always truncate down to the minute.

So for both, I agree with you that we don't need to worry right now and just try it out in Benefits.

**Review by thekaveman — APPROVED** [review]
Nice :+1:

**Pull request merged**